### PR TITLE
Migrate to GitHub Actions as PR gate

### LINKED ISSUE #75: Move CI to Github Actions
**Is this a BUG REPORT or FEATURE REQUEST?**:
FEATURE
**What happened**:
We need to migrate away from `travis-ci.org` since it is being reset.
Screen Shot 2021-02-04 at 2 07 23 PM

**What you expected to happen**:
Setup github actions as ci replacement. 
https://docs.github.com/en/actions/quickstart

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Addon Manager version
- Kubernetes version :
```
$ kubectl version -o yaml
```

**Other debugging information (if applicable)**:
- Addon status:
```
$ kubectl describe addon <addon-name>
```
- controller logs:
```
$ kubectl logs <addon-manager-pod>
```

### PR DESCRIPTION
**Pull request by kevdowney**
Fixes #75 

Migrate to GitHub Actions to gate PRs.

Signed-off-by: kevdowney <kevdowney@gmail.com>

### CODE REVIEW

**Comment by ccfishk** [comment]
Your unit-test is failing.

File: .github/workflows/pr-gate.yml [hunk 1]
```diff
@@ -0,0 +1,70 @@
+name: pr-gate
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
```
Comment (ccfishk) [inline .github/workflows/pr-gate.yml:7]: the PR could not be within the master branch

File: .github/workflows/pr-gate.yml [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+jobs:
+
+  unit-test:
+    if: github.repository_owner == 'keikoproj'
+    runs-on: ubuntu-latest
```
Comment (ccfishk) [inline .github/workflows/pr-gate.yml:13]: why only utuntu ?

**Review by ccfishk — CHANGES_REQUESTED** [review]
we should still favor integration/bdd tests higher than the gate coz the unit tests are sort of empty in this PR.

File: .github/workflows/pr-gate.yml [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set up Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.17
```
Comment (ccfishk) [inline .github/workflows/pr-gate.yml:20]: we can not stick to one go version, should provide compitibility.

File: .github/workflows/pr-gate.yml [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+      - name: Download Kubebuilder
+        run: |
+          curl -L -o kubebuilder_2.3.2_linux_amd64.tar.gz https://github.com/kubernetes-sigs/kubebuilder/releases/download/v2.3.2/kubebuilder_2.3.2_linux_amd64.tar.gz
+          tar -zxvf  kubebuilder_2.3.2_linux_amd64.tar.gz
+          sudo mv kubebuilder_2.3.2_linux_amd64 /usr/local/kubebuilder
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v1
```
Comment (ccfishk) [inline .github/workflows/pr-gate.yml:55]: Also, we need specify docker version.

File: .github/workflows/pr-gate.yml (hunk 2 above)
Comment (kevdowney) [reply .github/workflows/pr-gate.yml:13]: ubuntu is part of their hosted runners.
https://github.com/actions/virtual-environments#available-environments

File: .github/workflows/pr-gate.yml (hunk 1 above)
Comment (kevdowney) [reply .github/workflows/pr-gate.yml:7]: PR to master, is where we run this.

File: .github/workflows/pr-gate.yml (hunk 3 above)
Comment (kevdowney) [reply .github/workflows/pr-gate.yml:20]: What do you mean, we only build the controller on one version of Go

File: .github/workflows/pr-gate.yml (hunk 4 above)
Comment (kevdowney) [reply .github/workflows/pr-gate.yml:55]: No, that's not an option by default.
https://github.com/marketplace/actions/docker-setup-buildx#with-qemu

**Comment by kevdowney** [comment]
> we should still favor integration/bdd tests higher than the gate coz the unit tests are sort of empty in this PR.

We won't run BDDs on PRs, the `make test` actually runs a small environment that mimicks k8s and it verifies as BDD does. For a more complex gate we'll do that for Releases.

File: .github/workflows/pr-gate.yml [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+        with:
+          platforms: arm64,amd64
```
Comment (shailshah9) [inline .github/workflows/pr-gate.yml:57]: Do we want to include arm, with arm64 here?

Comment (kevdowney) [reply .github/workflows/pr-gate.yml:57]: We will only build for `arm64` and `amd64`

**Review by shailshah9 — APPROVED** [review]
LGTM

File: .github/workflows/pr-gate.yml (hunk 5 above)
Comment (kevdowney) [reply .github/workflows/pr-gate.yml:57]: Fyi. I created a new issue for pushing, here we're just runnig tests & builds. This issue is still TBD to address [Pushes and Multi-arch](https://github.com/keikoproj/addon-manager/issues/106).

**Pull request merged**